CMakeLists.txt

```diff
@@ -91,6 +91,9 @@ else()
 endif()
 message(STATUS "Build shared libs: ${BUILD_SHARED_LIBS}")
 
+option(ENABLE_ASAN "Enable build with AddressSanitizer" OFF)
```

Comment:
There are a couple more sanitizers available. Their enablement is all the same - passing value to `-fsanitize`.
Did you consider adding an option to enable address or thread or.. sanitizer to make it flexible?

Reply:
Yeah, UBSAN and TSAN will be done separately. Just want to pull request one at a time for now.